content/guides/general/inline-blocks/hero-to-block.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Converting Hero to a Block
+---
+
+## Step 5 - create component and template
+
+This simple inline editing configuration is a great start, but the end goal of this demo is to create a Home page built with [_Inline_ \*Blocks](<*[https://tinacms.org/docs/inline-editing/inline-blocks](https://tinacms.org/docs/inline-editing/inline-blocks)*>),\* so let's convert our Hero into a block.
+
+[Blocks](https://tinacms.org/docs/inline-editing/inline-blocks#creating-a-block) are made up of **a component** to render while editing and **a template** to configure defaults, add fields, and other required data. Let's make the _Hero Block Component_ first.
+
+**components/hero.js**
+
+```diff
+import React from 'react';
+import {
+  InlineTextarea,
+  InlineText,
++  BlocksControls,
+} from 'react-tinacms-inline';
+import '../styles/Hero.css';
+
+-export function Hero() {
++ export function Hero({index}) {
+  return (
++   <BlocksControls index={index}>
```

Comment:
What if we didn't make the `BlocksControls` part of the `Hero` component. 

The first `Hero` could be wrapped in an `<InlineGroup name="hero">` instead, and then we could define the `HeroBlock` in the same file:

**components/hero.js**
```tsx
export function Hero() {
  // ...
}

const template = { ... }

export HeroBlock = {
  template, 
  Component(props) {
    return <BlocksControles index={index.index}><Hero {...props} /></BlocksControls>
  }
}
```

The idea being that you can define components with inline fields and use them in blocks, groups, or just straight in the form.

Reply:
I like your thinking, the issue though is that at first there are no fields for the settings modal and the `InlineGroup` is useless without those extra fields. But it could go, make it a group when we add the inline fields, and then add settings modal fields, and _then_ convert to a block...

While I like the diversity in showing how you can define the `Component`, I think the first example should be the most simple and probably the most common way people will define blocks. But perhaps we could touch on this alternative way of defining the component in the later 'Add More Blocks' section?

Reply:
I just saw this.

What if the first data wasn't grouped under 'block' and you didn't need to use `InlineGroup` at all?